src/frontend/src/components/public/PIAFormTabs/review/index.tsx

```diff
@@ -69,6 +78,47 @@ const PIAReview = ({ printPreview }: IReviewProps) => {
     setReviewForm(pia.review);
   }, [pia.review]);
 
+  /**
+   * Update reviewForm.programArea when
+   * Part 4 Assessment(storing personal information tab) (PIDSOC), If Assessment of Disclosures Outside of Canada is filled out in PIA,
+   * ADM(Assistant Deputy Minister) is a preselected role
+   * and can not be deleted
+   * personalInformation.storedOutsideCanada is yes
+   * sensitivePersonalInformation.doesInvolve is yes
+   * sensitivePersonalInformation.disclosedOutsideCanada is no
+   */
+  useEffect(() => {
+    // if the condition does satisfy the rule, add the adm to programArea
+    // otherwise do nothing
+    if (
+      pia?.storingPersonalInformation?.personalInformation
+        ?.storedOutsideCanada === YesNoInput.YES &&
+      pia?.storingPersonalInformation?.sensitivePersonalInformation
+        .doesInvolve === YesNoInput.YES &&
+      pia?.storingPersonalInformation?.sensitivePersonalInformation
+        .disclosedOutsideCanada === YesNoInput.NO
+    ) {
+      setMandatoryADM(true);
+      if (
+        !pia?.review?.programArea.selectedRoles.includes(ApprovalRoles.ADM) &&
+        !reviewForm.programArea?.selectedRoles.includes(ApprovalRoles.ADM)
```

Comment:
why checking for both? 🤔

Reply:
!reviewForm.programArea?.selectedRoles.includes(ApprovalRoles.ADM) will move the second check